### DISCUSSION TITLE
Confusing naming for datetime components

### DISCUSSION THREAD

**Discussion by alex-kinokon** [palantir/blueprint#6530] [Q&A] [answered]
#### Question

As a library end user I am very confused by the current naming scheme for datetime components.

1. `Date*3` components are in the `datetime2` package, not a new `datetime3` package.
2. `Date*2` components are actually in `datetime` package, and not the `datetime2` package.
3. If I need to use `Date*3` components, I will need to import CSS from both `datetime2` and `datetime`.
4. In fact, the [docs say](https://github.com/palantir/blueprint/blob/7d12971bb8d8518c9b859f8e86ce6c790db96f74/packages/datetime2/src/index.md?plain=1#L23) `Date*3` components are “currently available” in the `datetime` package (?)

**Comment by adidahiya** [comment]
@alex-kinokon Sorry for that, you're right it's a bit confusing! We considered various options here for the naming scheme before landing on this one, if you're curious you can read [this PR thread](https://github.com/palantir/blueprint/pull/5935). This is a temporary state until we get to Blueprint v6.0.

> 4\. In fact, the [docs say](https://github.com/palantir/blueprint/blob/7d12971bb8d8518c9b859f8e86ce6c790db96f74/packages/datetime2/src/index.md?plain=1#L23) `Date*3` components are “currently available” in the `datetime` package (?)

This is a typo, I will fix it to say **@blueprintjs/datetime2** package.

**Comment by alex-kinokon** [comment]
@adidahiya Thanks. Is there any reason you don’t want to deprecate and freeze the `@blueprintjs/datetime2` package, and release the newer version under `@blueprintjs/datetime3` or `@blueprintjs/datetime-next` instead?

**Reply by adidahiya** [reply]
- There's an overhead to creating new packages and rolling those out, especially in large code bases;  it creates some extra code churn. So I'm not inclined to create a new package for the "V3" components.
- We could have named `@blueprintjs/datetime2` as `@blueprintjs/datetime-next` instead. Perhaps that is a better name. The components would have had the same names in such a package (DateInput, DateInput2, DateInput3, etc.). I will consider that package name for the next iteration of major API breaking changes. For now, you can just treat `@blueprintjs/datetime2` as if it was named `@blueprintjs/datetime-next`.

**Comment by ggdouglas** [comment, marked as answer]
Closing this out. The naming situation was resolved in Blueprint v6.

The modern date/time components now live in `@blueprintjs/datetime`. `@blueprintjs/datetime2` still exists for backwards compatibility, but it just re-exports from `@blueprintjs/datetime` with everything marked `@deprecated`. New code should import from `@blueprintjs/datetime` and ignore `datetime2`.
